src/Voron/Data/BTrees/Tree.cs

```diff
@@ -1205,26 +1205,34 @@ public List<long> AllPages()
                     {
                         if ((State.Flags & TreeFlags.FixedSizeTrees) == TreeFlags.FixedSizeTrees)
                         {
-                            var valueReader = GetValueReaderFromHeader(node);
-                            var valueSize = ((FixedSizeTreeHeader.Embedded*)valueReader.Base)->ValueSize;
-
-                            Slice fixedSizeTreeName;
-                            using (p.GetNodeKey(_llt, i, out fixedSizeTreeName))
+                            try
                             {
-                                var fixedSizeTree = new FixedSizeTree(_llt, this, fixedSizeTreeName, valueSize);
-
-                                var pages = fixedSizeTree.AllPages();
-                                results.AddRange(pages);
+                                var valueReader = GetValueReaderFromHeader(node);
+                                var valueSize = ((FixedSizeTreeHeader.Embedded*)valueReader.Base)->ValueSize;
 
-                                if ((State.Flags & TreeFlags.Streams) == TreeFlags.Streams)
+                                Slice fixedSizeTreeName;
+                                using (p.GetNodeKey(_llt, i, out fixedSizeTreeName))
                                 {
-                                    Debug.Assert(fixedSizeTree.ValueSize == ChunkDetails.SizeOf);
+                                    var fixedSizeTree = new FixedSizeTree(_llt, this, fixedSizeTreeName, valueSize);
+
+                                    var pages = fixedSizeTree.AllPages();
+                                    results.AddRange(pages);
+
+                                    if ((State.Flags & TreeFlags.Streams) == TreeFlags.Streams)
+                                    {
+                                        Debug.Assert(fixedSizeTree.ValueSize == ChunkDetails.SizeOf);
 
-                                    var streamPages = GetStreamPages(fixedSizeTree, GetStreamInfo(fixedSizeTree.Name, writable: false));
+                                        var streamPages = GetStreamPages(fixedSizeTree, GetStreamInfo(fixedSizeTree.Name, writable: false));
 
-                                    results.AddRange(streamPages);
+                                        results.AddRange(streamPages);
+                                    }
                                 }
                             }
+                            catch (InvalidFixedSizeTree)
```

Comment:
I don't understand the scenario in which this can happen or why we want to handle this?

Reply:
The condition we have above here (`if ((State.Flags & TreeFlags.FixedSizeTrees) == TreeFlags.FixedSizeTrees)`) is `true` for table trees because it contains FST indexes. 

Although table trees also contains regular entries (e.g. "Stats", "Schemas" records). So we can safely ignore that a node isn't FST tree header if root object type is `RootObjectType.Table`

Reply:
Can we do the check before  and not catch the exception?

Reply:
Yes, fixed in https://github.com/aviv86/ravendb/pull/1/commits/e3abdb178fb86b4addfba3a996ab02afd07399c5